rs/cross-chain/proposal-cli/src/dashboard/responses.rs

```diff
@@ -0,0 +1,44 @@
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeSet;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct CanisterInfo {
+    pub canister_id: String,
+    pub controllers: Vec<String>,
+    pub module_hash: String,
+    pub subnet_id: String,
+    pub upgrades: Option<Vec<CanisterUpgradeInfo>>,
+}
+
+impl CanisterInfo {
+    pub fn list_upgrade_proposals(&self) -> BTreeSet<u64> {
```

Comment:
Just out of curiosity, why not simply return a vector instead of a `BTreeSet`?

Reply:
Because like this results are already sorted 😎 . This is relevant because I need the last proposal, aka the one with the largest number. This also increases robustness because it doesn't rely on the ordering provided by the dashboard API which is a priori not guaranteed.

Reply:
Fee free to ping me if there is anything left opened. Marking as resolved to merge.